awx/main/models/notifications.py

```diff
@@ -201,56 +247,145 @@ def get_absolute_url(self, request=None):
+    JOB_FIELDS_WHITELIST = ['id', 'type', 'url', 'created', 'modified', 'name', 'description', 'job_type', 'inventory',
+                            'project', 'playbook', 'forks', 'limit', 'verbosity', 'job_tags', 'force_handlers', 'skip_tags',
+                            'start_at_task', 'timeout', 'use_fact_cache', 'unified_job_template', 'launch_type', 'status',
+                            'failed', 'started', 'finished', 'elapsed', 'job_explanation', 'execution_node', 'controller_node',
+                            'result_traceback', 'event_processing_finished', 'job_template', 'allow_simultaneous', 'scm_revision',
+                            'instance_group', 'diff_mode', 'job_slice_number', 'job_slice_count', 'custom_virtualenv',
+                            {'host_status_counts': ['skipped', 'ok', 'changed', 'failures', 'dark']},
+                            {'playbook_counts': ['play_count', 'task_count']},
+                            {'summary_fields': [{'inventory': ['id', 'name', 'description', 'has_active_failures',
+                                                               'total_hosts', 'hosts_with_active_failures', 'total_groups',
+                                                               'groups_with_active_failures', 'has_inventory_sources',
+                                                               'total_inventory_sources', 'inventory_sources_with_failures',
+                                                               'organization_id', 'kind']},
+                                                {'project': ['id', 'name', 'description', 'status', 'scm_type']},
+                                                {'project_update': ['id', 'name', 'description', 'status', 'failed']},
+                                                {'job_template': ['id', 'name', 'description']},
+                                                {'unified_job_template': ['id', 'name', 'description', 'unified_job_type']},
+                                                {'instance_group': ['name', 'id']},
+                                                {'created_by': ['id', 'username', 'first_name', 'last_name']},
+                                                {'labels': ['count', 'results']}]}]
+
+    @classmethod
+    def context_stub(cls):
+        """Returns a stub context that can be used for validating notification messages.
+        Context has the same structure as the context that will actually be used to render
+        a notification message."""
+        context = {'job': {},
+                   'job_friendly_name': 'stub',
+                   'url': 'stub'}
+
+        def build_context_for_fields(node, fields):
+            for f in fields:
+                if type(f) is dict:
+                    subname, subfields = f.copy().popitem()
+                    node[subname] = {}
+                    build_context_for_fields(node[subname], subfields)
+                else:
+                    node[f] = 'stub'
```

Comment:
This is neat, buuuuuuuut it might be nice to actually give a *real* example value for people when they test these notifications.  This is a bit misleading, because these fields have non-string types in practice:

![image](https://user-images.githubusercontent.com/214912/62067350-7e964700-b201-11e9-8dbc-58f24f66c1ef.png)

Reply:
Totally makes sense. The 'auto-generation' is nice, but inaccurate. Will post a static stub instead.

Reply:
Posted